cmd/tailscale/cli/network-lock.go

```diff
@@ -622,3 +624,48 @@ func runNetworkLockLog(ctx context.Context, args []string) error {
 	}
 	return nil
 }
+
+var nlTskeyWrapCmd = &ffcli.Command{
+	Name:       "tskey-wrap",
+	ShortUsage: "tskey-wrap <tailscale pre-auth key>",
+	ShortHelp:  "Modifies a pre-auth key from the admin panel to work with tailnet lock",
+	LongHelp:   "Modifies a pre-auth key from the admin panel to work with tailnet lock",
+	Exec:       runTskeyWrapCmd,
+}
+
+func runTskeyWrapCmd(ctx context.Context, args []string) error {
+	if len(args) != 1 {
+		return errors.New("usage: lock tskey-wrap <tailscale pre-auth key>")
+	}
+
+	st, err := localClient.StatusWithoutPeers(ctx)
+	if err != nil {
+		return fixTailscaledConnectError(err)
+	}
+
+	// Generate a separate tailnet-lock key just for the credential signature.
+	// We use the free-form meta strings to mark a little bit of metadata about this
+	// key.
+	priv := key.NewNLPrivate()
+	k := tka.Key{
+		Kind:   tka.Key25519,
+		Public: priv.Public().Verifier(),
+		Votes:  1,
+		Meta: map[string]string{
+			"purpose":            "pre-auth key",
```

Comment:
might want to include the non-secret part (stableid) of the key

Reply:
I had this originally, but this then means we no longer treat the 'authkey' part of the wrapped authkey as an opaque string, which limits its format for compatibility for future changes.

Maybe we can embed a cryptographic hash of the authkey? thoughts?

Reply:
@willnorris Do you think its likely we'll majorly revamp the authkey format in the near future?

Maybe if I:
1. Check that it has a `tskey-auth-` prefix
2. Check that it has another `-` and the stableID bit ends in `CNTRL`

its fairly safe to embed the stableID bit into the generated tailnet lock key?

Reply:
> @willnorris Do you think its likely we'll majorly revamp the authkey format in the near future?
> 
> Maybe if I:
> 
> 1. Check that it has a `tskey-auth-` prefix
> 2. Check that it has another `-` and the stableID bit ends in `CNTRL`
> 
> its fairly safe to embed the stableID bit into the generated tailnet lock key?

Don't assume a particular suffix on the stableID, that will break for compliance-US and onprem.

Reply:
I suspect that tskey-{type}-{stableid} is unlikely to change anytime soon, so that should be a safe way to extract the stable id (until of course, it isn't).  But as Adrian said, certainly don't assume CTRL.

Reply:
Maybe verifying the prefix is enough, I think catching obvious mistakes has value but I agree that expecting a particular format is problematic and I don't like the idea of asking control about an auth key unless it's being used to auth. I guess having an auth key validity oracle API wouldn't be too bad but we'd need to be careful with the secret part and definitely never send the wrapper.

Oh, one thing it should reject is if the key is already wrapped.

Reply:
Done & done!